### PR TITLE
Removed fullscreen prop in modal and use size instead

### PR DESCRIPTION
**Pull request by wakooka**

### CODE REVIEW

**Comment by grosbouddha** [comment]
Make sure you do the kairos integration quickly after merging this one. Thanks !

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Unreleased
 
-- [...]
+- **[BREAKING CHANGE]** - Replaced `fullscreen` prop from `Modal` with new entry in enum `ModalSize`
```
Comment (mcellier) [inline CHANGELOG.md]: by a new entry?

Comment (wakooka) [reply CHANGELOG.md]: nope `with` is the correct way to say it in english :)

File: src/_utils/assert.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { SuccessModalSize } from 'successModal/SuccessModal'
+import { ConfirmationModalSize } from 'confirmationModal/ConfirmationModal'
+
+export const assertModalSizes = (
+  modalSize: typeof ModalSize | typeof SuccessModalSize | typeof ConfirmationModalSize,
+  size: string,
+): void => {
+  const correctModalSize = Object.values(modalSize).includes(size)
+  if (!correctModalSize) {
+    throw new Error(`This modal only accepts modal size: ${JSON.stringify(Object.keys(modalSize))}`)
```
Comment (mcellier) [inline src/_utils/assert.ts]: You're repeating yourself. "This modal only accept the following sizes:"

File: src/_utils/assert.unit.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { ConfirmationModalSize } from 'confirmationModal/ConfirmationModal'
+
+describe('assertModalSizes', () => {
+  it('Should throw if modal size is not in the allowed size list', () => {
+    expect(() => assertModalSizes(ConfirmationModalSize, 'fullscreen')).toThrowError(
+      'This modal only accepts modal size: ["SMALL","LARGE"]',
+    )
+  })
+  it('Should return undefined when', () => {
+    expect(assertModalSizes(ConfirmationModalSize, 'large')).toBeUndefined()
```
Comment (mcellier) [inline src/_utils/assert.unit.ts]: Use `ModalSize.LARGE`

File: src/_utils/assert.unit.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { assertModalSizes } from './assert'
+import { ConfirmationModalSize } from 'confirmationModal/ConfirmationModal'
+
+describe('assertModalSizes', () => {
+  it('Should throw if modal size is not in the allowed size list', () => {
+    expect(() => assertModalSizes(ConfirmationModalSize, 'fullscreen')).toThrowError(
```
Comment (mcellier) [inline src/_utils/assert.unit.ts]: Use ModalSize.FULLSCREEN

File: src/confirmationModal/ConfirmationModal.tsx [hunk 5]
```diff
@@ -8,20 +8,34 @@ import { ModalProps } from 'modal/Modal'
...
+  SMALL = 'small',
+  LARGE = 'large',
+}
+
 export interface ConfirmationModalProps extends ModalProps {
   readonly status: ConfirmationModalStatus
   readonly onConfirm?: () => void
   readonly confirmLabel?: string
   readonly confirmIsLoading?: boolean
+  readonly size?: ModalSize
```
Comment (mcellier) [inline src/confirmationModal/ConfirmationModal.tsx]: ConfirmationModalSize?

File: src/confirmationModal/ConfirmationModal.tsx [hunk 6]
```diff
@@ -8,20 +8,34 @@ import { ModalProps } from 'modal/Modal'
...
 }
 
 class ConfirmationModal extends Component<ConfirmationModalProps> {
+  static defaultProps: Partial<ConfirmationModalProps> = {
+    onConfirm() {},
+    confirmLabel: '',
+    confirmIsLoading: false,
+    size: ModalSize.SMALL,
```
Comment (mcellier) [inline src/confirmationModal/ConfirmationModal.tsx]: ConfirmationModalSize.SMALL ?

File: src/confirmationModal/ConfirmationModal.tsx (hunk 5 above)
Comment (wakooka) [reply src/confirmationModal/ConfirmationModal.tsx]: No, cause this props is passed down to <Modal>

File: src/confirmationModal/ConfirmationModal.tsx (hunk 6 above)
Comment (wakooka) [reply src/confirmationModal/ConfirmationModal.tsx]: No cause this props is passed down to `<Modal>`

**Pull request merged**